### PR TITLE
[LUNA-1675]: [BpkSegmentedControls] fixes focus styling

### PR DESCRIPTION
**Pull request by LouiseReid**
**Before**

https://github.com/user-attachments/assets/c11ad7ab-0a07-4044-88bf-a72493cfc8da

**After**

https://github.com/user-attachments/assets/ef13c175-2b6a-4dcc-8ede-3b43a574c186

Remember to include the following changes:

- [x] Ensure the PR title includes the name of the component you are changing so it's clear in the release notes for consumers of the changes in the version e.g `[KOA-123][BpkButton] Updating the colour`
- [ ] `README.md` (If you have created a new component)
- [ ] Component `README.md`
- [ ] Tests
- [ ] Accessibility tests
    - The following checks were performed:
        - [ ] Ability to navigate using a [keyboard only](https://webaim.org/techniques/keyboard/)
        - [ ] Zoom functionality ([Deque University explanation](https://dequeuniversity.com/checklists/web/text)):
            - [ ] The page SHOULD be functional AND readable when only the text is magnified to 200% of its initial size
            - [ ] Pages must reflow as zoom increases up to 400% so that content continues to be presented in only one column i.e. Content MUST NOT require scrolling in two directions (both vertically and horizontally)
        - [ ] Ability to navigate using a [screen reader only](https://webaim.org/articles/screenreader_testing/)
- [ ] Storybook examples created/updated
- [ ] For breaking changes or deprecating components/properties, migration guides added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here

### CODE REVIEW

File: packages/bpk-component-segmented-control/src/BpkSegmentedControl.module.scss [hunk 1]
```diff
@@ -102,3 +102,19 @@
...
+.bpk-segmented-control:last-child {
+  border-top-right-radius: tokens.$bpk-border-radius-sm;
+  border-bottom-right-radius: tokens.$bpk-border-radius-sm;
+}
+
+.bpk-segmented-control:focus {
+  border: (tokens.$bpk-one-pixel-rem * 2) solid tokens.$bpk-text-link-day;
+  outline: none;
+  outline-offset: (tokens.$bpk-one-pixel-rem * 2);
+}
```
Comment (LouiseReid) [inline packages/bpk-component-segmented-control/src/BpkSegmentedControl.module.scss:120]: Setting borders the outer children to ensure the focus isn't clipped off the corners.

Adding outline none is to ensure this extra outline doesn't show

Screenshot 2024-07-16 at 14 57 13

File: packages/bpk-component-segmented-control/src/BpkSegmentedControl.module.scss [hunk 2]
```diff
@@ -102,3 +102,19 @@
...
+
+.bpk-segmented-control:last-child {
+  border-top-right-radius: tokens.$bpk-border-radius-sm;
+  border-bottom-right-radius: tokens.$bpk-border-radius-sm;
+}
+
+.bpk-segmented-control:focus {
+  border: (tokens.$bpk-one-pixel-rem * 2) solid tokens.$bpk-text-link-day;
+  outline: none;
+  outline-offset: (tokens.$bpk-one-pixel-rem * 2);
```
Comment (olliecurtis) [inline packages/bpk-component-segmented-control/src/BpkSegmentedControl.module.scss:119]: We have `bpk-focus-indicator()` to support this :) 

https://github.com/Skyscanner/backpack/blob/main/packages/bpk-mixins/_utils.scss#L34-L37

Comment (olliecurtis) [reply packages/bpk-component-segmented-control/src/BpkSegmentedControl.module.scss:119]: You will just have to add the `outline: none` yourself

Comment (LouiseReid) [reply packages/bpk-component-segmented-control/src/BpkSegmentedControl.module.scss:119]: I was getting errors when trying to use it that it was undefined even though utils is imported at the top of the file. Let me try again

Comment (olliecurtis) [reply packages/bpk-component-segmented-control/src/BpkSegmentedControl.module.scss:119]: The use should look something like 

`@include utils.bpk-focus-indicator`

Comment (LouiseReid) [reply packages/bpk-component-segmented-control/src/BpkSegmentedControl.module.scss:119]: That doesn't work as the mixin sets the outline to be `(tokens.$bpk-one-pixel-rem * 2) solid tokens.$bpk-text-link-day;` where as we need it to be the border here

Comment (LouiseReid) [reply packages/bpk-component-segmented-control/src/BpkSegmentedControl.module.scss:119]: The outline offset here wasn't actually doing anything

**Review by amburrrshand — APPROVED** [review]
